source/server/options_impl.cc

```diff
@@ -30,6 +31,24 @@
 #endif
 
 namespace Envoy {
+
+const Stats::Tag parseTag(const std::string tag) {
```

Comment:
Could consider making this a constructor for `Stats::Tag`?

Reply:
No, in my opnion. This parsing rule is only for CLI flags so it's not good to place this in the constructor. If another use case for this parsing rule happens, then we can move this into a builder method or object of `Stats::Tag`. I think it's not time to do that yet.